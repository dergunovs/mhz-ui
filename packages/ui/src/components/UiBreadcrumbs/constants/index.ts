export const LINKS = [
  { url: '/', title: 'Main' },
  { url: '/category', title: 'Categories' },
  { url: '/cpu', title: 'CPU' },
];

export const DEFAULT_COLOR = 'default';
export const DEFAULT_ARIA_LABEL = 'Breadcrumb';
