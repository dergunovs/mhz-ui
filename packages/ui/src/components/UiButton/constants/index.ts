export const DEFAULT_LAYOUT = 'primary';
export const DEFAULT_TYPE = 'button';
export const DEFAULT_ICON = undefined;

export const DEFAULT_SLOT = 'Button label';
export const DEFAULT_ARIA_LABEL = 'Button';
