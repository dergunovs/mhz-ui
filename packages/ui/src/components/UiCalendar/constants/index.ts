import { ICalendarEvent } from '../interface';

export const EVENTS: ICalendarEvent<object>[] = [
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 1, text: 'Text' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 36%, rgb(235, 0, 123) 69%, rgb(251, 185, 0) 100%)',
  },
  {
    id: 'id of event',
    start: new Date(),
    end: new Date(),
    title: '+',
    content: [{ id: 2, text: 'Text 2' }],
    color: 'linear-gradient(135deg, rgb(168, 60, 255) 41%, rgb(235, 0, 123) 71%, rgb(0, 134, 255) 100%)',
  },
];
