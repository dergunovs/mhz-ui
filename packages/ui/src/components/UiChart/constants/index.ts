export const LABELS = [
  '07.07 - 13.07',
  '14.07 - 20.07',
  '21.07 - 27.07',
  '28.07 - 03.08',
  '04.08 - 10.08',
  '11.08 - 17.08',
  '18.08 - 24.08',
];

export const DATASETS = [
  {
    data: [28, 16, 12, 4, 22, 16, 14],
    label: 'Грудь',
    borderColor: '#00CED1',
    backgroundColor: '#00CED1',
  },
  {
    data: [4, 6, 0, 2, 2, 0, 6],
    label: 'Ноги',
    borderColor: '#000080',
    backgroundColor: '#000080',
  },
  {
    data: [22, 18, 8, 10, 25, 18, 16],
    label: 'Плечи',
    borderColor: '#C41E3A',
    backgroundColor: '#C41E3A',
  },
  {
    data: [0, 6, 0, 6, 5, 4, 2],
    label: 'Пресс',
    borderColor: '#32CD32',
    backgroundColor: '#32CD32',
  },
  {
    data: [32, 24, 12, 10, 33, 24, 20],
    label: 'Руки',
    borderColor: '#DA70D6',
    backgroundColor: '#DA70D6',
  },
  {
    data: [16, 18, 8, 12, 26, 22, 20],
    label: 'Спина',
    borderColor: '#FF8C00',
    backgroundColor: '#FF8C00',
  },
];

export const TYPE = 'Line';
export const TITLE = 'Chart title';
