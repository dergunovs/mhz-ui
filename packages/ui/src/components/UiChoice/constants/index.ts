export const options = [1, 2, 3, 4, 5];
export const modelValue = options[2];
export const title = 'Заголовок';
export const isTall = true;
export const isInput = true;
