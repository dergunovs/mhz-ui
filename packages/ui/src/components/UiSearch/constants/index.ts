export const DEBOUNCE_TIME = 300;

export const MODEL_VALUE = '';
export const LANG = 'ru';
export const IS_SUCCESS = true;

export const RESULTS = {
  products: [
    { _id: '1', title: 'AMD Ryzen 7700X' },
    { _id: '2', title: 'AMD Ryzen 7900X' },
  ],
  manufacturers: [{ _id: '1', title: 'AMD' }],
};

export const SEARCH_SCHEME = [
  { type: 'products', labels: ['title'], url: '/products' },
  { type: 'manufacturers', labels: ['title'], url: '/manufacturers' },
];
