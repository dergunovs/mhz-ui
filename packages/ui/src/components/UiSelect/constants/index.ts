export const MODEL_VALUE = { _id: '1', title: 'Один', title_en: 'One' };

export const IS_FILTER = false;

export const OPTIONS = ['1', '2', '3', '4', '5', '6', '7', '8', '11', '111', '12'];

export const OPTIONS_OBJECTS = [
  { _id: '1', title: 'Один', title_en: 'One' },
  { _id: '2', title: 'Два', title_en: 'Two' },
  { _id: '3', title: 'Три', title_en: 'Three' },
];
