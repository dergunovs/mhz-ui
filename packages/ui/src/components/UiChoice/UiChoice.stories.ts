import { Meta, StoryObj } from '@storybook/vue3';
import { useArgs } from 'storybook/preview-api';

import { isTall, modelValue, options, title, isInput } from './constants';

import { html } from '@/utils';
import { UiChoice } from '@/components';

const meta: Meta<typeof UiChoice> = {
  component: UiChoice,
  args: {
    isTall,
    modelValue,
    options,
    title,
    isInput,
  },
  parameters: {
    docs: {
      description: {
        component: '',
      },
    },
  },
};

const argTypes = {};

type Story = StoryObj<typeof UiChoice>;

export default meta;

export const Primary: Story = {
  render: (args, { updateArgs }) => ({
    components: { UiChoice },
    setup: () => ({ args, argTypes, updateArgs }),

    template: html` <UiChoice v-bind="args" @update:modelValue="update" />`,

    methods: {
      update(value: number) {
        updateArgs({ modelValue: value });
      },
    },
  }),

  decorators: [
    (story, context) => {
      const [args, updateArgs] = useArgs();

      return story({ ...context, updateArgs, args });
    },
  ],

  argTypes,
};
