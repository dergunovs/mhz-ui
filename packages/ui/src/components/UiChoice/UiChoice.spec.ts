import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import UiChoice from './UiChoice.vue';

import { isTall, modelValue, options, title, isInput } from './constants';
import { wrapperFactory } from '@/test';

const titleText = dataTest('ui-choice-title');
const buttons = dataTest('ui-choice-buttons');
const button = dataTest('ui-choice-button');
const input = dataTest('ui-choice-input');

let wrapper: VueWrapper<InstanceType<typeof UiChoice>>;

beforeEach(() => {
  wrapper = wrapperFactory(UiChoice, { options, modelValue, title, isTall, isInput });
});

enableAutoUnmount(afterEach);

describe('UiChoice', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(UiChoice)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows title', async () => {
    expect(wrapper.find(titleText).text()).toBe(title);
  });

  it('shows tall buttons by props', async () => {
    expect(wrapper.find(buttons).attributes('data-tall')).toBe(isTall.toString());

    await wrapper.setProps({ isTall: false });

    expect(wrapper.find(buttons).attributes('data-tall')).toBe('false');
  });

  it('shows buttons', async () => {
    expect(wrapper.findAll(button).length).toBe(options.length);
    expect(wrapper.find(button).text()).toBe(options[0].toString());
    expect(wrapper.find(button).attributes('data-current')).toBe((modelValue === options[0]).toString());
  });

  it('shows input', async () => {
    expect(wrapper.find(input).attributes('value')).toBe(modelValue.toString());
  });

  it('emits choosen value', async () => {
    await wrapper.find(button).trigger('click');

    expect(wrapper.emitted('update:modelValue')).toHaveLength(1);
    expect(wrapper.emitted()['update:modelValue'][0]).toStrictEqual([options[0]]);

    wrapper.find(input).setValue(options[1]);

    expect(wrapper.emitted('update:modelValue')).toHaveLength(2);
    expect(wrapper.emitted()['update:modelValue'][1]).toStrictEqual([options[1]]);
  });
});
