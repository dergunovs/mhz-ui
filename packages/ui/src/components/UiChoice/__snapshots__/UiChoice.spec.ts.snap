// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiChoice > matches snapshot 1`] = `
"<div class="_container_bde654"><label data-test="ui-choice-title">Заголовок</label>
  <div class="_options_bde654" data-tall="true" data-test="ui-choice-buttons"><button type="button" class="_button_bde654" data-current="false" data-fill="false" data-test="ui-choice-button">1</button><button type="button" class="_button_bde654" data-current="false" data-fill="false" data-test="ui-choice-button">2</button><button type="button" class="_button_bde654" data-current="true" data-fill="false" data-test="ui-choice-button">3</button><button type="button" class="_button_bde654" data-current="false" data-fill="false" data-test="ui-choice-button">4</button><button type="button" class="_button_bde654" data-current="false" data-fill="false" data-test="ui-choice-button">5</button><input type="number" class="_input_bde654" step="1" min="1" max="60" data-test="ui-choice-input" value="3"></div>
</div>"
`;
